### PR TITLE
Gets rid of both ballistic gun fowling* and ballistic emp fulnerability

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

Shrimple as the title, removes gun fowling and thus jamming from every gun except for some bolt actions, as those bolt actions can fowl and jam on tg so I've just reverted them to their standard behavior. Also gets rid of the whole thing with being able to emp certains guns to do... ???

## How This Contributes To The Skyrat Roleplay Experience

Both of these have reasons to go.

Guns getting dirty and jamming will either literally never be a problem, or will be such a problem that it makes a gun useless. It also really isn't all that engaging of a mechanic to go "you've fired your gun x times now its useless until you use a rag on muahahah!" nor does such a mechanic add anything meaningful to fights.

Guns being vulnerable to emps was definitely a ??? moment. The entire thing about guns is that emps don't do anything to them, so why we felt the need to have emps fuck with only certain guns but not others is just confusing and much like fowling, adds nothing meaningful to fights outside of the emp implant being usable against more than it should be.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
Unbelievable batman I can shoot a gun for more than 2 seconds without it being unusable, and I can emp it because it has nothing electronic in it. Amazing!

image

image

  
</details>

## Changelog

:cl:
del: Guns getting dirty over time due to use has been reverted to tg behavior, that is to say literally only surplus mosins and maint muskets have it now.
del: Guns no longer have situational emp vulnerability/invulnerability. EMPing a ballistic gun won't do anything to it.
/:cl:

### CODE REVIEW

**Comment by Paxilmaniac** [comment]
HOW BRO THIS IS ON MASTER FROM 15 MINUTES AGO

**Comment by RimiNosha** [comment]
image

**Comment by Paxilmaniac** [comment]
Its technically balance yes but literally what does gbp mean the bot can't touch me.

**Comment by ghost** [comment]
Okay Im all for removing both but can I please be able to clean the guns even if it does nothing please I need to have mechanics to tie into my ocd roleplay 🥺

**Comment by RatFromTheJungle** [comment]
remove it from everything, 'tg default' or not; rid us of the plague

(it's never been fun, it just adds random steps to specific guns, that no other gun has)

**Comment by Paxilmaniac** [comment]
Jungle the skyrat system for it is completely gone literally only the maint muskets and the shitty surplus mosins have 'dirt' anymore because they always have, all I did was reset the file back to just be 1:1 with tg

**Comment by RatFromTheJungle** [comment]
image
image
this is what forcing pax to touch gunsprites has done to them 😔 

this PR adds it back to two guns that didn't have it before; while removing it from guns that aren't available as of currently ((the armadyne stuff, at which point i'd just point at the reasons you listed above in the pr desc for why it isn't a good idea

> Jungle the skyrat system for it is completely gone literally only the maint muskets and the shitty surplus mosins have 'dirt' anymore because they always have, all I did was reset the file back to just be 1:1 with tg

**Comment by Paxilmaniac** [comment]
literally what are you talking about jungle, the mosins always had the cobalt gun dirt system, as well as a vast majority of the guns that guncargo sold

**Comment by Paxilmaniac** [comment]
> this PR adds it back to two guns that didn't have it before;

image

immeasurable cap bro you can just tell me you're upset about the mosin getting nerfed

**Comment by RatFromTheJungle** [comment]
> literally what are you talking about jungle, the mosins always had the cobalt gun dirt system, as well as a vast majority of the guns that guncargo sold

as of right now, go on server and run like, ten stripper clips through it; I can assure you, it hasn't in ages, because 'realistic' never did anything for bolt-actions

**Comment by RatFromTheJungle** [comment]
> literally what are you talking about jungle, the mosins always had the cobalt gun dirt system, as well as a vast majority of the guns that guncargo sold

as of right now, go on server and run like, ten stripper clips through it; I can assure you, it hasn't in ages, because 'realistic' never really did anything for bolt-actions; as compared to other guns, and, skimming the TG code, it's a 5% gain per shot; which, again:

> Guns getting dirty and jamming will either literally never be a problem, or will be such a problem that it makes a gun useless. It also really isn't all that engaging of a mechanic to go "you've fired your gun x times now its useless until you use a rag on muahahah!" nor does such a mechanic add anything meaningful to fights.

seems a bit off-mark for the given reasons; if you're taking it away from stuff (we don't really use at the minute), to be specifically returning a worse-off version of the exact problem, to two guns in specific, which we do use


https://www.youtube.com/watch?v=SYA6RLLZ-PA

**Comment by YakumoChen** [comment]
guns being weak to EMPs is more a product on the proliferation of ballistics to keep EMPs being a valid way to stymie security, given on TG they only typically have energy weapons (weak to EMPs).

I'm kinda eh on this, since ballistics don't really _have_ a counter like energy guns do, so it's still almost always preferable to use a ballistic weapon.

**Comment by Paxilmaniac** [comment]
i merged the pr that made this conflict

File: code/modules/projectiles/guns/ballistic/rifle.dm [hunk 1]
```diff
@@ -59,7 +59,11 @@
 	knife_x_offset = 27
 	knife_y_offset = 13
 	can_be_sawn_off = TRUE
-	realistic = TRUE
+	var/jamming_chance = 20
```
Comment (Zonespace27) [inline code/modules/projectiles/guns/ballistic/rifle.dm:62]: so like, was this just some undocumented SR removal in the first place

Edit: yes it was

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Review by Zonespace27 — APPROVED** [review]
Way I see it, mosins and similar are intentionally shit (while still being obtainable ~~croon~~) and it's a "good enough" reason to either pack a rag, more mosins, or pray. Having it for every gun is pretty 🤷 considering how it'll only be a problem after hundreds of rounds, at which point it becomes a game-stopper if you weren't prepped

**Pull request merged**